### ISSUE TITLE
workflow error: node acompcor failed to run on host 

### ISSUE THREAD

**Issue by EricDeveaud** [nipreps/fmriprep#2761]
### What happened?

ACompCor nodes are failing

using fmriprep-20.0.1 singularity container built as described in the doc.

### What command did you use?

```shell
`fmriprep ds000005/ tmp participant 2>&1 | tee log`

I can provide the full log, or any other file, if required
```

### What version of fMRIPrep are you running?

fMRIPrep v21.0.1

### How are you running fMRIPrep?

Singularity

### Is your data BIDS valid?

Yes

### Are you reusing any previously computed results?

No

### Please copy and paste any relevant log output.

```shell
Node: fmriprep_wf.single_subject_01_wf.func_preproc_task_mixedgamblestask_run_01_wf.bold_confounds_wf.rename_acompcor
Working directory: /home/gensoft/work/fmriprep_wf/single_subject_01_wf/func_preproc_task_mixedgamblestask_run_01_wf/bold_confounds_wf/rename_acompcor

Node inputs:

components_file = <undefined>
metadata_file = <undefined>

Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node
    result["result"] = node.run(updatehash=updatehash)
  File "/opt/conda/lib/python3.8/site-packages/nipype/pipeline/engine/nodes.py", line 516, in run
    result = self._run_interface(execute=True)
  File "/opt/conda/lib/python3.8/site-packages/nipype/pipeline/engine/nodes.py", line 635, in _run_interface
    return self._run_command(execute)
  File "/opt/conda/lib/python3.8/site-packages/nipype/pipeline/engine/nodes.py", line 741, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/opt/conda/lib/python3.8/site-packages/nipype/interfaces/base/core.py", line 428, in run
    runtime = self._run_interface(runtime)
  File "/opt/conda/lib/python3.8/site-packages/fmriprep/interfaces/confounds.py", line 139, in _run_interface
    components = pd.read_csv(self.inputs.components_file, sep='\t')
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers.py", line 610, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers.py", line 462, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers.py", line 819, in __init__
    self._engine = self._make_engine(self.engine)
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers.py", line 1050, in _make_engine
    return mapping[engine](self.f, **self.options)  # type: ignore[call-arg]
  File "/opt/conda/lib/python3.8/site-packages/pandas/io/parsers.py", line 1873, in __init__
    self._reader = parsers.TextReader(self.handles.handle, **kwds)
  File "pandas/_libs/parsers.pyx", line 521, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file
```

### Additional information / screenshots

command ran on ds000005 data grabbed from https://gin.g-node.org/nipreps-data/fmriprep-test-data

as I'm not at all confident with what I'm doing it may be some bad input from my part ;-)
anyway seems related to #2659. closed some days ago (7 as typing) that I guess whas not included in 20.0.1 release from january 24 

NB answered tht bids are valid ones, but if a `not sure` option would have been available, I would have checked this one ;-)
anyway bids-validator just gave some warnings about 
```
1: [WARN] You should define 'SliceTiming' for this file. If you don't provide this information slice time correction will not be possible. 'Slice Timing' is the time at which each slice was acquired within each volume (frame) of the acquisition. Slice timing is not slice order -- rather, it is a list of times containing the time (in seconds) of each slice acquisition in relation to the beginning of volume acquisition. (code: 13 - SLICE_TIMING_NOT_DEFINED)
                ./sub-01/func/sub-01_task-mixedgamblestask_run-01_bold.nii.gz
                ./sub-01/func/sub-01_task-mixedgamblestask_run-02_bold.nii.gz
 ```
                
NB2 bids validator embeded in the container is version 1.8.0 while upstream repo have 1.9.3 

regards

Eric

**Comment by mgxd** [comment]
Looks like you encountered the same problem we've been sporadically running into when testing - interestingly I haven't seen it affect single echo data before. My guess is there some injected randomness that is affecting the BOLD brain mask and causing this - if you try rerunning, do you still run into the error?

**Comment by EricDeveaud** [comment]
yes second run gave the same error.

```
	 [Node] Finished "fmriprep_wf.single_subject_01_wf.func_preproc_task_mixedgamblestask_run_02_wf.bold_std_trans_wf.bold_reference_wf.enhance_and_skullstrip_bold_wf.apply_mask".
220413-11:52:51,948 nipype.workflow ERROR:
	 could not run node: fmriprep_wf.single_subject_01_wf.func_preproc_task_mixedgamblestask_run_01_wf.bold_confounds_wf.rename_acompcor
220413-11:52:51,952 nipype.workflow ERROR:
	 could not run node: fmriprep_wf.single_subject_01_wf.func_preproc_task_mixedgamblestask_run_02_wf.bold_confounds_wf.rename_acompcor
220413-11:52:51,963 nipype.workflow CRITICAL:
	 fMRIPrep failed: Workflow did not execute cleanly. Check log for details
220413-11:52:52,843 cli ERROR:
	 Preprocessing did not finish successfully. Errors occurred while processing data from participants: 01 (2). Check the HTML reports for details.
```

and it is again  `pandas.errors.EmptyDataError: No columns to parse from file`

**Comment by dlevitas** [comment]
I noticed this too in #2702 but will post here since this is an open issue. From what I can tell, this arises from a poor brain mask generated in the SDC workflow.
![bad_brain_mask](https://user-images.githubusercontent.com/25011343/163582991-06c225e5-9273-4f8c-9d46-05e98a000ca6.PNG)

**Comment by mgxd** [comment]
@dlevitas Hopefully this has been fixed with the changes in `21.0.2` - can you give that version a try?

**Comment by dlevitas** [comment]
@mgxd version `21.0.2` alleviates the issue for me, thanks!
